ios-base/Helpers/Errors.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  Errors.swift
+//  ios-base
+//
+//  Created by German on 5/20/19.
+//  Copyright © 2019 TopTier labs. All rights reserved.
+//
+
+import Foundation
+
+/**
+ App-specific error domains enumerated.
+ Includes a default error code and a default a localized description.
+ */
+enum ErrorDomain {
+  case generic
+  case parsing(_: Localizable.Type)
+  case networkRequest
+  
+  var name: String {
+    switch self {
+    case .parsing: return "PARSING_ERROR"
+    case .networkRequest: return "NETWORK_FAILURE"
+    default: return "GENERIC_ERROR"
+    }
+  }
+  
+  var defaultCode: ErrorCode {
+    switch self {
+    case .parsing: return .invalidData
+    case .networkRequest: return .badNetworkRequest
+    default: return .generic
+    }
+  }
+  
+  var localizedDescription: String {
+    switch self {
+    case .parsing(let objectClass):
+      return String(format: name.localized, objectClass.localized)
+    default: return name.localized
+    }
+  }
+}
+
+enum ErrorCode: Int {
+  case generic = 18190
+  case invalidData = 18191
+  case badNetworkRequest = 400
+}
+
+protocol Localizable {
+  static var localizeKey: String { get }
+  static var localized: String { get }
+}
+
+extension Localizable {
+  static var localizeKey: String {
+    return String(describing: self)
+  }
+  
+  static var localized: String { return localizeKey.localized }
+}
+
+extension App {
```

Comment:
why not making this an Error extension?

Reply:
I started with that idea but I needed this `NSError` initializer. Also I think stating `App.error()` gives you a better the idea and is less verbose.